utils/slither_format/format_external_function.py

```diff
@@ -0,0 +1,55 @@
+import re, logging
+from slither.utils.colors import red, yellow, set_colorization_enabled
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger('Slither.Format')
+set_colorization_enabled(True)
+
+class FormatExternalFunction:
+
+    @staticmethod
+    def format (slither, patches, elements):
+        for element in elements:
+            Found = False
+            for contract in slither.contracts:
+                if not Found and contract.name == element['type_specific_fields']['parent']['name']:
+                    for function in contract.functions:
+                        if function.name == element['name']:
+                            # If function parameters are written to in function body then we cannot convert this function
+                            # to external because external function parameters are allocated in calldata region which is
+                            # non-modifiable. See https://solidity.readthedocs.io/en/develop/types.html#data-location
+                            if not FormatExternalFunction.function_parameters_written(function):
```

Comment:
This should be removed as we have now https://github.com/crytic/slither/pull/257

Reply:
Will remove once #257 is merged on `dev` else it fails unit tests.